src/lib.rs

```diff
@@ -294,7 +297,11 @@ impl InstructionOperands {
                     .map(|source| InstructionOperands::WAIT {
                         polarity: o0 >> 2,
                         source,
-                        index: o1,
+                        index: if source == WaitSource::IRQ {
+                            o1 & 7
```

Comment:
nit: Can we use `0b111` here ? I find it more readable/obvious about the intent to mask the lower bits (rather than some other properties of the integer 7).